test/conformance/enqueue/urEnqueueUSMMemcpy.cpp

```diff
@@ -1,2 +1,140 @@
+        }
+        if (device_src) {
+            EXPECT_SUCCESS(urUSMFree(context, device_src));
+        }
+        if (device_dst) {
+            EXPECT_SUCCESS(urUSMFree(context, device_dst));
+        }
+
+        UUR_RETURN_ON_FATAL_FAILURE(urQueueTest::TearDown());
+    }
+
+    bool verifyData() {
+        EXPECT_SUCCESS(
+            urEnqueueUSMMemcpy(queue, true, host_mem.data(), device_dst, allocation_size, 0, nullptr, nullptr));
+        return std::all_of(host_mem.begin(), host_mem.end(), [this](uint8_t i) { return i == memset_value; });
+    }
+
+    const uint32_t num_elements = 1024;
+    const uint8_t memset_value = 12;
+    const uint32_t allocation_size = sizeof(uint8_t) * num_elements;
+    std::vector<uint8_t> host_mem = std::vector<uint8_t>(num_elements);
+
+    ur_event_handle_t memset_event = nullptr;
+    int *device_src{nullptr};
+    int *device_dst{nullptr};
+};
+
+/**
+ * Test that urEnqueueUSMMemcpy blocks when the blocking parameter is set to true.
+ */
+TEST_P(urEnqueueUSMMemcpyTest, Blocking) {
+
+    ASSERT_SUCCESS(urEventWait(1, &memset_event));
+    ASSERT_SUCCESS(urEnqueueUSMMemcpy(queue, true, device_dst, device_src, allocation_size, 0, nullptr, nullptr));
+    ASSERT_TRUE(verifyData());
+}
+
+/**
+ * Test that urEnqueueUSMMemcpy blocks and returns an event with UR_EVENT_STATUS_COMPLETE when the blocking
+ * parameter is set to true.
+ */
+TEST_P(urEnqueueUSMMemcpyTest, BlockingWithEvent) {
+
+    ur_event_handle_t memcpy_event = nullptr;
+    ASSERT_SUCCESS(urEventWait(1, &memset_event));
+    ASSERT_SUCCESS(
+        urEnqueueUSMMemcpy(queue, true, device_dst, device_src, allocation_size, 0, nullptr, &memcpy_event));
+
+    ur_event_status_t event_status;
+    ASSERT_SUCCESS(urEventGetInfo(memcpy_event, UR_EVENT_INFO_COMMAND_EXECUTION_STATUS, sizeof(ur_event_status_t),
+                                  &event_status, nullptr));
+    ASSERT_EQ(event_status, UR_EVENT_STATUS_COMPLETE);
+    EXPECT_SUCCESS(urEventRelease(memcpy_event));
+    ASSERT_TRUE(verifyData());
+}
+
+/**
+ * Test that the memory copy happens when the blocking flag is set to false and
+ * the application waits for the returned event to complete.
+ */
+TEST_P(urEnqueueUSMMemcpyTest, NonBlocking) {
+
+    ASSERT_SUCCESS(urEventWait(1, &memset_event));
+    ur_event_handle_t memcpy_event = nullptr;
+    ASSERT_SUCCESS(
+        urEnqueueUSMMemcpy(queue, false, device_dst, device_src, allocation_size, 0, nullptr, &memcpy_event));
+    ASSERT_SUCCESS(urEventWait(1, &memcpy_event));
+
+    ASSERT_TRUE(verifyData());
+}
+
+/**
+ * Test that urEnqueueUSMMemcpy waits for the events dependencies before copying the memory.
+ */
+TEST_P(urEnqueueUSMMemcpyTest, WaitForDependencies) {
+
+    ASSERT_SUCCESS(
+        urEnqueueUSMMemcpy(queue, true, device_dst, device_src, sizeof(int), 1, &memset_event, nullptr));
+    ASSERT_TRUE(verifyData());
+}
+
+TEST_P(urEnqueueUSMMemcpyTest, InvalidNullQueueHandle) {
+
```

Comment:
What happens to the `memset_event` in these tests?

Reply:
It gets released on TearDown()

Reply:
I mean does it mater that you don't wait for the event to complete with `urEventWait` like you've done in the previous tests?

Reply:
I don't think it matters. I'm not expecting a successful memory copy in negative tests. So the fact that the memset might not be complete is irrelevant to the outcome of the test.

Reply:
Sounds good. I wasn't sure if you had to wait for the event to complete before releasing it 👍